### PR TITLE
chore: supporting python 3.11

### LINKED ISSUE #213: Update sqlacodegen to 3.0.0rc3
### Describe the problem

There is an incompatibility notice for python 3.11 https://github.com/MobilityData/mobility-feed-api#python-311-incompatibility-notice

### Proposed solution

The issue seems to be resolved with the latest release candidate of sqlacodegen

### Alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by davidgamez**
## Major Changes
- Upgrade Python packages to support `~=3.11` version. This helps to overcome the previous incompatibility issue with 3.11 from sqlcodegen library.
- Upgrade Pydantic to 2.x version. This version has a few breaking changes. The most noticeable is the `Optional` parameters are now `mandatory` but can be set to None. The following functions's signature:  `f (x: Optional[Y])`, is now equals to `f (x: Y | None)`. More info [here](https://github.com/pydantic/pydantic?tab=readme-ov-file#pydantic-v110-vs-v2).
- Open API generator upgrade to 7.50. This adds support to enums code generation and a few other enhancements, more info [here](https://github.com/OpenAPITools/openapi-generator/releases).
- Update GitHub actions

Closes #213 

## Testing tips:

- Checkout branch
- Update your Python environment to 3.11 version
- Install the new libraries
```
pip3 install -r api/requirements.txt
pip3 install -r api/requirements_dev.txt
```
- Reset your local DB(**Warning this will delete all your local data**)
```
./scripts/docker-localdb-rebuild-data.sh --populate-db
```
- Generate API models
```
scripts/api-gen.sh
```
- Generate DB models
```
scripts/db-gen.sh
```
- Execute local tests:
```
./scripts/api-tests.sh --folder api
./scripts/api-tests.sh --folder functions-python
```
- Start the application
```
scripts/api-start.sh
```
- Test endpoints locally, example queries:
```
curl --location 'http://localhost:8080/v1/gtfs_rt_feeds?\
--header 'Accept: application/json'
```
```
curl --location 'http://localhost:8080/v1/gtfs_feeds\
--header 'Accept: application/json'
```

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `./scripts/api-tests.sh` to make sure you didn't break anything
- [x] Add or update any needed documentation to the repo
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

File: api/src/feeds/impl/models/bounding_box_impl.py [hunk 1]
```diff
@@ -14,7 +14,6 @@ class Config:
         Enabling `from_orm` method to create a model instance from a SQLAlchemy row object."""
 
         from_attributes = True
-        orm_mode = True
```
Comment (davidgamez) [inline api/src/feeds/impl/models/bounding_box_impl.py:17]: This parameter was decommissioned in favor of `from_attributes`.

File: docs/DatabaseCatalogAPI.yaml [hunk 2]
```diff
@@ -253,14 +253,9 @@ paths:
                   results:
                     type: array
                     items:
-                      oneOf:
```
Comment (davidgamez) [inline docs/DatabaseCatalogAPI.yaml:256]: New openapi codegen generates classes that changed the signature of the response. The new signature uses the discriminator, which can be handy in a few cases but is less intuitive to consume.

File: scripts/gen-config.yaml [hunk 3]
```diff
@@ -2,3 +2,5 @@
 additionalProperties:
   packageName: feeds_gen
   fastapiImplementationPackage: feeds.impl
+  # Adding this commented line for future reference as it is not currently supported by the fastApi generator
```
Comment (davidgamez) [inline scripts/gen-config.yaml:5]: Even the `legacyDiscriminatorBehavior` is listed in the documentation. It's not fully supported by the fastApi generator.

**Comment by davidgamez** [comment]
This PR should be merged after #383.

File: api/requirements_dev.txt [hunk 4]
```diff
@@ -1,7 +1,7 @@
 #Testing
 pytest>=3.2.3
-pytest_mock>=3.11.1
 flake8==6.0.0
-black==23.3.0
+black==22.3.0
```
Comment (davidgamez) [inline api/requirements_dev.txt:4]: This was necessary to keep runtime packages aligned(click and fastapi).

**Comment by davidgamez** [comment]
This PR will be ready for review after #489

**Review by cka-y — APPROVED** [review]
LGTM 🥳 Any plan to also move python cloud functions to 3.11?

**Comment by davidgamez** [comment]
> LGTM 🥳 Any plan to also move python cloud functions to 3.11?

Good point, we should let's create an issue for it

**Pull request merged**